### PR TITLE
Extending Watch Api consistency. Postgres Buffer size fix. 

### PR DESCRIPTION
**Pull request by staskosatkin**
PR linked to issue https://github.com/authzed/spicedb/issues/1357

There is added buffer length checking before sending to the chan. So Watch API could continue streaming data. The one limit is GC period duration.

Please let me know if you have any concerns about this PR

### CODE REVIEW

**Comment by josephschorr** [comment]
@staskosatkin Can you please sign the CLI?

**Comment by staskosatkin** [comment]
**_I have read the CLA Document and I hereby sign the CLA_**

**Comment by staskosatkin** [comment]
Hi @josephschorr ! Could you please take a look?

**Comment by josephschorr** [comment]
@staskosatkin Apologies, I was offline for a few days. Will take a look tomorrow

File: internal/datastore/crdb/watch.go [hunk 1]
```diff
@@ -119,6 +120,17 @@ func (cds *crdbDatastore) Watch(ctx context.Context, afterRevision datastore.Rev
 				})
 
 				for _, change := range toEmit {
+					for int(cds.watchBufferLength) <= len(updates) {
```
Comment (josephschorr) [inline internal/datastore/crdb/watch.go:123]: So I'm not convinced this is necessary; if we remove the `default` clause from the select below, then the `updates <- change` should hang if the update channel is filled.

Can you try that approach?

Comment (josephschorr) [reply internal/datastore/crdb/watch.go:123]: @staskosatkin

Comment (josephschorr) [reply internal/datastore/crdb/watch.go:123]: Correct; the channel insert will hang, which is what we expect, correct?

Comment (staskosatkin) [reply internal/datastore/crdb/watch.go:123]: I will check this approach and update this PR after that. Thank you.

Comment (josephschorr) [reply internal/datastore/crdb/watch.go]: @staskosatkin Checking in on this

Comment (josephschorr) [reply internal/datastore/crdb/watch.go]: @staskosatkin Checking in again

**Comment by josephschorr** [comment]
Closing due to lack of movement; Please feel to reopen if necessary

**Pull request closed without merging**